nixos/maintainers/scripts/lxd/lxd-virtual-machine-image-inner.nix

```diff
@@ -14,7 +14,18 @@
     ];
 
   networking.useDHCP = false;
-  networking.interfaces.eth0.useDHCP = true;
+
+  systemd.network = {
+    enable = true;
+    networks."50-eth0" = {
+      matchConfig.Name = "eth0";
+      networkConfig = {
+        DHCP = "ipv4";
```

Comment:
(note that with useNetworkd the shorthand will generate networkd configuration instead of network script)

Reply:
> why just ra for ipv6? systemd can do dhcp for both and fallback to ra for v6

Seems pretty uncommon to do stateful DHCPv6, but I can enable it, if Incus supports that.

> also can we use   networking.interfaces.eth0.useDHCP = true; instead of explicitly defining everything? or is this shorthand deperacted?

The maintenance situation of `networking.interfaces` options is questionable at best. If we can write the thing in plain networkd, we should.

Reply:
https://www.freedesktop.org/software/systemd/man/latest/systemd.network.html#DHCP=

Networkd will start a DHCPv6 client, if the RA tells it to. I think that is sufficient.

Reply:
ah, ok. that should be sufficient.